### PR TITLE
Create codeql-analysis.yml

### PR DESCRIPTION
**Pull request by ryjones**
#### Type of change

- Infrastructure update

#### Description

Initial commit of codeql scanner settings.

### CODE REVIEW

File: .github/workflows/codeql-analysis.yml [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    - name: Checkout repository
+      uses: actions/checkout@v2
+      with:
+        # We must fetch at least the immediate parents so that if this is
+        # a pull request then we can checkout the head.
+        fetch-depth: 2
+
+    # If this run was triggered by a pull request event, then checkout
+    # the head of the pull request instead of the merge commit.
+    - run: git checkout HEAD^2
```
Comment (denyeart) [inline .github/workflows/codeql-analysis.yml:28]: Why checkout prior commit? Don't we want to analyze the PR commit?

What if the PR has multiple commits, HEAD^2 wouldn't be correct in that case...

**Review by denyeart — COMMENTED** [review]
What does the scan result "Missing analysis for base commit" mean?  Do we know what the scan results are, prior to enabling this?

File: .github/workflows/codeql-analysis.yml (hunk 1 above)
Comment (ryjones) [reply .github/workflows/codeql-analysis.yml:28]: The merge would be HEAD, HEAD~1 should be the one before that - @btl5037 ?

Comment (ryjones) [reply .github/workflows/codeql-analysis.yml:28]:
```
Run git checkout HEAD^2

Previous HEAD position was c36d1bf 

Merge 6739eb0013f82e8be3346daf78fbcff3c3b88a22 into af0d871b3b3cf9915ed023c0d013b1ef0721ef12

HEAD is now at 6739eb0 Create codeql-analysis.yml```

looks right to me

File: .github/workflows/codeql-analysis.yml [hunk 2]
```diff
@@ -0,0 +1,55 @@
+# SPDX-License-Identifier: Apache-2.0
+name: "CodeQL"
+
+on:
+  push:
+    branches: [master, release-1.0, release-1.1, release-1.2, release-1.3, release-1.4, release-2.0, release-2.1, release-2.2]
+  pull_request:
+    # The branches below must be a subset of the branches above
+    branches: [master]
```
Comment (sykesm) [inline .github/workflows/codeql-analysis.yml:9]: Why every pull request instead of just the merges?

File: .github/workflows/codeql-analysis.yml [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+    # ✏️ If the Autobuild fails above, remove it and uncomment the following three lines
+    #    and modify them (or add more) to build your code if your project
+    #    uses a compiled language
+
+    #- run: |
+    #   make bootstrap
+    #   make release
+
+    - name: Perform CodeQL Analysis
+      uses: github/codeql-action/analyze@v1
```
Comment (sykesm) [inline .github/workflows/codeql-analysis.yml:55]: This is running `make` for no good reason and it seems to be ignoring the failure.

```
  [2020-07-27 21:04:12] [build-err] 2020/07/27 21:04:12 Makefile found, running make
```
followed by
```
  [2020-07-27 21:11:49] [build-err] fatal: ambiguous argument 'HEAD^..HEAD': unknown revision or path not in the working tree.
  [2020-07-27 21:11:49] [build-err] Use '--' to separate paths from revisions, like this:
  [2020-07-27 21:11:49] [build-err] 'git <command> [<revision>...] -- [<file>...]'
  [2020-07-27 21:11:49] [build] All files are excluded from having license headers
  [2020-07-27 21:11:49] [build] Building github.com/client9/misspell/cmd/misspell -> misspell
  [2020-07-27 21:11:49] [build-err] go: downloading github.com/client9/misspell v0.3.4
  [2020-07-27 21:11:51] [build-err] fatal: ambiguous argument 'HEAD^..HEAD': unknown revision or path not in the working tree.
  [2020-07-27 21:11:51] [build-err] Use '--' to separate paths from revisions, like this:
  [2020-07-27 21:11:51] [build-err] 'git <command> [<revision>...] -- [<file>...]'
  [2020-07-27 21:11:51] [build] Checking changed go files for spelling errors ...
```
and
```
  [2020-07-27 21:12:04] [build-err] ./scripts/golinter.sh: line 29: goimports: command not found
  [2020-07-27 21:12:04] [build-err] make: *** [linter] Error 127
  [2020-07-27 21:12:04] [build] Makefile:176: recipe for target 'linter' failed
  [2020-07-27 21:12:04] [build-err] 2020/07/27 21:12:04 Running /usr/bin/make failed, continuing anyway: exit status 2
```

File: .github/workflows/codeql-analysis.yml [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+name: "CodeQL"
+
+on:
+  push:
+    branches: [master, release-1.0, release-1.1, release-1.2, release-1.3, release-1.4, release-2.0, release-2.1, release-2.2]
+  pull_request:
+    # The branches below must be a subset of the branches above
+    branches: [master]
+  schedule:
+    - cron: '0 5 * * 6'
```
Comment (sykesm) [inline .github/workflows/codeql-analysis.yml:11]: Why on commit *and* on interval?

**Review by sykesm — CHANGES_REQUESTED** [review]
It would be good to see some security scans but I don't think what's here is what we want.

If we're going to do it for pull requests, I'd expect it to be across all branches and not just master. That said, this sort of thing could also be part of the pre-release process.

I'm also concerned about why a failing build is initiated by the tool and then ignored. Since we vendor all of our dependencies and we don't generate code during build, no `make` should be required.

**Comment by sykesm** [comment]
@ryjones I'm going to close this for now. I haven't seen any comments on the feedback or changes to the PR. Please reopen if you want to revisit this.

Thanks.

**Pull request closed without merging**